workflow-controller/controller-service/src/main/java/io/hyscale/controller/validator/impl/LoadBalancerValidator.java

```diff
@@ -0,0 +1,166 @@
+import io.hyscale.servicespec.commons.model.service.ServiceSpec;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.stereotype.Component;
+
+import java.util.ArrayList;
+import java.util.List;
+
+@Component
+public class LoadBalancerValidator implements Validator<WorkflowContext> {
+
+    private static final Logger logger = LoggerFactory.getLogger(LoadBalancerValidator.class);
+
+    /**
+     * Checks for the required mandatory fields.
+     * Checks for Ports mismatch
+     * Checks for external true.
+     * @param workflowContext
+     * @return
+     * @throws HyscaleException
+     */
+    @Override
+    public boolean validate(WorkflowContext workflowContext) throws HyscaleException {
+        logger.debug("Validating load balancer details from the service spec");
+        ServiceSpec serviceSpec = workflowContext.getServiceSpec();
+        TypeReference<LoadBalancer> loadBalancerTypeReference = new TypeReference<LoadBalancer>() {
+        };
+
+        LoadBalancer loadBalancer = serviceSpec.get(HyscaleSpecFields.loadBalancer, loadBalancerTypeReference);
+        if (loadBalancer != null) {
+            if (!validateMandatoryFields(loadBalancer)) {
+                return false;
+            }
+            if (!portValidation(serviceSpec, loadBalancer)) {
+                //message for validator?
+                return false;
+            }
+            if (!checkForExternalTrue(serviceSpec)) {
+                return false;
+            }
+        }
+        return true;
+    }
+
+    /**
+     * Checks for Ports mismatch
+     * @param serviceSpec
+     * @param loadBalancer
+     * @return
+     * @throws HyscaleException
+     */
+    private boolean portValidation(ServiceSpec serviceSpec, LoadBalancer loadBalancer) throws HyscaleException {
+        TypeReference<List<Port>> portsListTypeReference = new TypeReference<List<Port>>() {
+        };
+        List<Port> portList = serviceSpec.get(HyscaleSpecFields.ports, portsListTypeReference);
+        List<String> portNumbersList = new ArrayList<>();
+        portList.forEach(each -> portNumbersList.add(each.getPort()));
+        List<String> lbPorts = new ArrayList<>();
+        loadBalancer.getMapping().forEach(e -> lbPorts.add(e.getPort()));
+        for (String lbPort : lbPorts) {
+            if (!portNumbersList.contains(lbPort)) {
+                WorkflowLogger.persist(ValidatorActivity.PORTS_MISMATCH, LoggerTags.ERROR, lbPort);
+                return false;
+            }
+        }
+        return true;
+    }
+
+
+    /**
+     * @param serviceSpec
+     * @throws HyscaleException
+     */
+    private boolean checkForExternalTrue(ServiceSpec serviceSpec) throws HyscaleException {
+        TypeReference<Boolean> booleanTypeReference = new TypeReference<>() {
+        };
+        Boolean isExternal = serviceSpec.get(HyscaleSpecFields.external, booleanTypeReference);
+        if (isExternal != null && !isExternal) {
```

Comment:
No need for TypeReference object, you can directly specify Boolean.class
In if statement you can use BooleanUtils.isFalse